pkg/controller/multiarchbuildconfig/multiarchbuildconfig.go

```diff
@@ -61,32 +64,25 @@ func AddToManager(mgr manager.Manager, architectures []string, dockerCfgPath str
 		UpdateFunc:  func(event.UpdateEvent) bool { return true },
 		GenericFunc: func(event.GenericEvent) bool { return false },
 	}
-	if err := c.Watch(source.Kind(mgr.GetCache(), &v1.MultiArchBuildConfig{}), mabcHandler(), predicateFuncs); err != nil {
+
+	if err := c.Watch(source.Kind(mgr.GetCache(), &v1.MultiArchBuildConfig{}),
+		&handler.EnqueueRequestForObject{}, predicateFuncs); err != nil {
 		return fmt.Errorf("failed to create watch for MultiArchBuildConfig: %w", err)
 	}
 
 	return nil
 }
 
-func mabcHandler() handler.EventHandler {
-	return handler.EnqueueRequestsFromMapFunc(func(ctx context.Context, o ctrlruntimeclient.Object) []reconcile.Request {
-		mabc, ok := o.(*v1.MultiArchBuildConfig)
-		if !ok {
-			logrus.WithField("type", fmt.Sprintf("%T", o)).Error("Got object that was not a MultiArchBuildConfig")
-			return nil
-		}
-
-		return []reconcile.Request{
-			{NamespacedName: types.NamespacedName{Namespace: mabc.Namespace, Name: mabc.Name}},
-		}
-	})
-}
-
 type reconciler struct {
-	logger         *logrus.Entry
-	client         ctrlruntimeclient.Client
-	architectures  []string
-	manifestPusher manifestpusher.ManifestPusher
+	logger           *logrus.Entry
+	client           ctrlruntimeclient.Client
+	architectures    []string
+	manifestPusher   manifestpusher.ManifestPusher
+	dockerConfigPath string
+
+	// For testing purpose only
+	timeNowFn      func() time.Time
+	mirrorImagesFn func(log *logrus.Entry, registryConfig string, images []string) error
```

Comment:
Also this.

Reply:
Adding something in the reconciling struct just for the test cases is smelly. It can be done by another approach or the structure of the tested function should be improved.

Reply:
In the end I need a way to somehow mock [mirrorImages](https://github.com/openshift/ci-tools/pull/3676/files#diff-da2cf6dcc4c106cde64397ab1fa87a001d400aae2e5fd563afadefa98ac8cf43R63). I could pass the function to [handleMirrorImage](https://github.com/openshift/ci-tools/pull/3676/files#diff-da2cf6dcc4c106cde64397ab1fa87a001d400aae2e5fd563afadefa98ac8cf43R27) like so:
```golang
type mirrorFunc func(log *logrus.Entry, registryConfig string, images []string) error
// and then add it as a new arg here
func (r *reconciler) handleMirrorImage(srcImage string, mabc *v1.MultiArchBuildConfig, mirror mirrorFunc) (func(mabcToMutate *v1.MultiArchBuildConfig), bool) { ... }
```
but then the problem raises again when we unit test the `reconcile` function:
```golang
// again here...
func (r *reconciler) reconcile(ctx context.Context, req reconcile.Request, logger *logrus.Entry, mirror mirrorFunc) error { ... }
```
I really would like to avoid this cascade effect as it's smelly as well.

What about an interface like the [manifestPusher](https://github.com/openshift/ci-tools/blob/fb0bf5d51faaff2760e5033ba3788314dd22c1af/pkg/controller/multiarchbuildconfig/multiarchbuildconfig.go#L80)?

Reply:
Yea we can do a similar approach.